airbyte-integrations/connectors/destination-jdbc/src/main/java/io/airbyte/integrations/destination/jdbc/copy/CopyConsumer.java

```diff
@@ -116,21 +127,30 @@ public void close(boolean hasFailed) throws Exception {
   }
 
   public void closeAsOneTransaction(List<StreamCopier> streamCopiers, boolean hasFailed, JdbcDatabase db) throws Exception {
+    Exception firstException = null;
     try {
       StringBuilder mergeCopiersToFinalTableQuery = new StringBuilder();
       for (var copier : streamCopiers) {
-        copier.closeStagingUploader(hasFailed);
-
-        if (!hasFailed) {
-          copier.createTemporaryTable();
-          copier.copyStagingFileToTemporaryTable();
-          copier.createDestinationSchema();
-          var destTableName = copier.createDestinationTable();
-          var mergeQuery = copier.generateMergeStatement(destTableName);
-          mergeCopiersToFinalTableQuery.append(mergeQuery);
+        try {
```

Comment:
Not blocking here, but generally it feels better to have visibility on an independent bug fix like this in the master commit messages.

"prevent snowflake s3 staging hanging on record-level failures" at that level would make it clear for someone running into this problem if/when a fix was introduced. They wouldn't be able to match to this title based on their problem.